### PR TITLE
feat(rpc): implement Filecoin.EthSyncing

### PR DESCRIPTION
**Pull request by hanabi1224**
## Summary of changes

Changes introduced in this pull request:

- implement `Filecoin.EthSyncing` RPC method

Lotus code:
```go
func (a *EthModule) EthSyncing(ctx context.Context) (ethtypes.EthSyncingResult, error) {
	state, err := a.SyncAPI.SyncState(ctx)
	if err != nil {
		return ethtypes.EthSyncingResult{}, fmt.Errorf("failed calling SyncState: %w", err)
	}

	if len(state.ActiveSyncs) == 0 {
		return ethtypes.EthSyncingResult{}, errors.New("no active syncs, try again")
	}

	working := -1
	for i, ss := range state.ActiveSyncs {
		if ss.Stage == api.StageIdle {
			continue
		}
		working = i
	}
	if working == -1 {
		working = len(state.ActiveSyncs) - 1
	}

	ss := state.ActiveSyncs[working]
	if ss.Base == nil || ss.Target == nil {
		return ethtypes.EthSyncingResult{}, errors.New("missing syncing information, try again")
	}

	res := ethtypes.EthSyncingResult{
		DoneSync:      ss.Stage == api.StageSyncComplete,
		CurrentBlock:  ethtypes.EthUint64(ss.Height),
		StartingBlock: ethtypes.EthUint64(ss.Base.Height()),
		HighestBlock:  ethtypes.EthUint64(ss.Target.Height()),
	}

	return res, nil
}

func (sr EthSyncingResult) MarshalJSON() ([]byte, error) {
	if sr.DoneSync {
		// when done syncing, the json response should be '"result": false'
		return []byte("false"), nil
	}

	// need to do an anonymous struct to avoid infinite recursion
	return json.Marshal(&struct {
		StartingBlock EthUint64 `json:"startingblock"`
		CurrentBlock  EthUint64 `json:"currentblock"`
		HighestBlock  EthUint64 `json:"highestblock"`
	}{
		StartingBlock: sr.StartingBlock,
		CurrentBlock:  sr.CurrentBlock,
		HighestBlock:  sr.HighestBlock})
}

func (e EthUint64) MarshalJSON() ([]byte, error) {
	return json.Marshal(e.Hex())
}
```

## Reference issue to close (if applicable)

Closes

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: src/lotus_json/mod.rs [hunk 1]
```diff
@@ -330,7 +330,7 @@ pub mod hexify {
         T: Num + std::fmt::LowerHex,
         S: Serializer,
     {
-        serializer.serialize_str(format!("0x{value:x}").as_str())
+        serializer.serialize_str(format!("{value:#x}").as_str())
```
Comment (hanabi1224) [inline src/lotus_json/mod.rs:333]: https://doc.rust-lang.org/stable/std/fmt/trait.LowerHex.html

Comment (elmattic) [reply src/lotus_json/mod.rs:333]: We could do same in `eth_block_number`.

Comment (hanabi1224) [reply src/lotus_json/mod.rs:333]: Fixed.

**Review by lemmih — APPROVED** [review]

File: src/rpc_api/mod.rs [hunk 2]
```diff
@@ -706,6 +708,78 @@ pub mod eth_api {
         }
     }
 
+    #[derive(Debug, Clone, Default)]
+    pub struct EthSyncingResult {
+        pub done_sync: bool,
+        pub startingblock: i64,
+        pub currentblock: i64,
+        pub highestblock: i64,
```
Comment (LesnyRumcajs) [inline src/rpc_api/mod.rs:716]: Is there a reason for not using `snake_case`?

Comment (hanabi1224) [reply src/rpc_api/mod.rs:716]: To match lotus and save serde rename lines. : )

Comment (LesnyRumcajs) [reply src/rpc_api/mod.rs:716]: We should use `serde` renames for consistency; otherwise, in some places, we may even use CamelCase...

Comment (LesnyRumcajs) [reply src/rpc_api/mod.rs:716]: Not a blocker, though.

**Review by LesnyRumcajs — APPROVED** [review]

File: src/rpc_api/mod.rs (hunk 2 above)
Comment (hanabi1224) [reply src/rpc_api/mod.rs]: @LesnyRumcajs Fixed. Please re-approve

**Review by LesnyRumcajs — APPROVED** [review]

**Pull request merged**